test/OrchardCore.Tests/Localization/LocalizationManagerTests.cs

```diff
@@ -67,11 +71,49 @@ public void GetDictionarySelectsPluralRuleFromProviderWithHigherPriority()
                 It.IsAny<CultureDictionary>())
             );
 
-            var manager = new LocalizationManager(new[] { _pluralRuleProvider.Object, highPriorityRuleProvider.Object }, new[] { _translationProvider.Object }, _memoryCache);
+            var manager = new LocalizationManager([_pluralRuleProvider.Object, highPriorityRuleProvider.Object], [_translationProvider.Object], _memoryCache);
 
             var dictionary = manager.GetDictionary(CultureInfo.GetCultureInfo("cs"));
 
             Assert.Equal(dictionary.PluralRule, csPluralRuleOverride);
         }
+
+
+        [Theory]
+        [InlineData("en", "Hello en !")]
+        [InlineData("zh-CN", "你好！")]
+        public async Task TestLocalizationRule(string culture, string expected)
+        {
+            var context = new SiteContext();
+            await context.InitializeAsync();
+            await context.UsingTenantScopeAsync(async scope =>
+            {
+                var shellFeaturesManager = scope.ServiceProvider.GetRequiredService<IShellFeaturesManager>();
+                var availableFeatures = await shellFeaturesManager.GetAvailableFeaturesAsync();
+                var featureIds = new string[] { "OrchardCore.Localization.ContentLanguageHeader", "OrchardCore.Localization" };
+                var features = availableFeatures.Where(feature => featureIds.Contains(feature.Id));
+                await shellFeaturesManager.EnableFeaturesAsync(features, true);
+                var siteService = scope.ServiceProvider.GetRequiredService<ISiteService>();
+                var siteSettings = await siteService.LoadSiteSettingsAsync();
+                siteSettings.Alter<LocalizationSettings>("LocalizationSettings", localizationSettings =>
+                {
+                    localizationSettings.DefaultCulture = culture;
+                    localizationSettings.SupportedCultures = [culture];
+                });
+                await siteService.UpdateSiteSettingsAsync(siteSettings);
+                var shellSettings = scope.ServiceProvider.GetRequiredService<ShellSettings>();
+                var shellHost = scope.ServiceProvider.GetRequiredService<IShellHost>();
+                await shellHost.ReleaseShellContextAsync(shellSettings);
+            });
+
+            await context.UsingTenantScopeAsync(scope =>
+            {
+                var S = scope.ServiceProvider.GetRequiredService<IStringLocalizer<LocalizationManagerTests>>();
+                
+                Assert.Equal(expected, S["hello!"]);
```

Comment:
@sebastienros  No problem, 

but how can I switch the current culture in this unit test?

Reply:
You can use `CultureScope`